src/proxy/BaseProxy.ts

```diff
@@ -25,8 +25,8 @@ export class BaseProxy {
         return deferred.promise;
     }
 
-    protected encodeInvokeOnKey<T>(codec: any, key: any, ...codecArguments: any[]): Q.Promise<T> {
-        var partitionId: number = this.client.getPartitionService().getPartitionId(key);
+    protected encodeInvokeOnKey<T>(codec: any, partitionKey: any, ...codecArguments: any[]): Q.Promise<T> {
+        var partitionId: number = this.client.getPartitionService().getPartitionId(partitionKey);
```

Comment:
This is actually not `partitionKey` but `key`. I mean the key from key-value pair. Please revert this naming change.